pkg/client/clientcmd/client_config_test.go

```diff
@@ -162,7 +162,12 @@ func TestCreateCleanWithPrefix(t *testing.T) {
 		{"anything.com:8080", "anything.com:8080", ""},
 		{"anything.com", "anything.com", ""},
 		{"anything", "anything", ""},
-		{"", "http://localhost:8080", ""},
+	}
+
+	if EnvVarCluster.Server != "" {
+		tt = append(tt, struct{ server, host, prefix string }{"", EnvVarCluster.Server, ""})
```

Comment:
How about calling
```
os.UnsetEnv("KUBERNETES_MASTER")
```
before running these tests, and explicitly testing the EnvVarCluster behavior in a separate test by calling
```
os.SetEnv("KUBERNETES_MASTER")
```
? That way the test is unaffected by local environment, but we still get coverage of the envvar handling.